### DISCUSSION TITLE
Replace mqtt-azure example with azure-sdk-for-c

### DISCUSSION THREAD

**Discussion by xiongyu0523** [zephyrproject-rtos/zephyr#37688] [General]
The current mqtt-azure example project on Zephyr is implemented according to MQTT protocol documented by Microsoft. While Microsoft now provide a very lightweight c sdk (helper) library to handle MQTT protocol stuff and allow user to bring their own connectivity stack (TLS, MQTT, TCP/IP) which should be good fit for Zephyr network, would you mind if I submit a PR to replace the original mqtt-azure project and document with the new one? Thanks.

**Comment by xiongyu0523** [comment]
Add Link:
https://github.com/Azure/azure-sdk-for-c

**Comment by aurel32** [comment]
MQTT is not a protocol specific to Microsoft, it has been created many years ago by IBM. The protocol being a standard there is no point in using one external implementation per provider.

**Reply by xiongyu0523** [reply]
I think you misunderstand my point. The proposal is to keep using Zephyr's native mqtt and network stack. Instead of implementing simple azure iot protocol at application level, Microsoft now provide a Bring-Your-Own-Stack c sdk take care of things on top of (ANY) MQTT implementation. I would like to update the original mqtt-azure example to use azure-sdk-for-c, which is considered reliable and can be used in commercial project
